test/compiler/codegen.jl

```diff
@@ -781,3 +781,7 @@ f_isdefined_nospecialize(@nospecialize(x)) = isdefined(x, 1)
 # Test codegen for isa(::Any, Type)
 f_isa_type(@nospecialize(x)) = isa(x, Type)
 @test !occursin("jl_isa", get_llvm(f_isa_type, Tuple{Any}, true, false, false))
+
+# Issue #47247
+f47247(a::Ref{Int64}, b::Nothing) = setfield!(a, :x, b)
+@test_throws TypeError f47247(Ref(5), nothing)
```

Comment:
needs whitespace (will fail the CI check--check your editor configuration to make it automatic)